### DISCUSSION TITLE
Where is "Consistent SPH Rigid-Fluid Coupling"?

### DISCUSSION THREAD

**Discussion by entropylost** [InteractiveComputerGraphics/SPlisHSPlasH#328] [Q&A] [answered]
The features list on the website "https://splishsplash.physics-simulation.org/features/" says that

> The SPlisHSPlasH library implements the following boundary handling methods:
> * Jan Bender, Lukas Westhofen, Stefan Rhys Jeske, “Consistent SPH Rigid-Fluid Coupling”, Vision, Modeling and Visualization, 2023. https://doi.org/10.2312/vmv.20231244

But I can't seem to find it anywhere, and it isn't documented in the other pages either.

If the code could be provided or that would be removed, that would be appreciated (the former would be significantly more preferred for obvious reasons).

**Comment by janbender** [comment]
The method is implemented in the DFSPH pressure solver.

**Reply by entropylost** [reply]
@janbender I don't see it, unless its Akinci2012? The only boundary methods you have in the BoundaryHandlingMethods enum are Akinci2012, Koschier2017, and Bender2019, and you don't seem to have a default case anywhere in the DFSPH.

I'm looking at https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/blob/master/SPlisHSPlasH/DFSPH/TimeStepDFSPH.cpp for reference, it could be somewhere else I guess.

**Comment by janbender** [comment]
It's a special implementation of Akinci, Koschier and Bender. 

Am 20. Juli 2024 07:17:04 MESZ schrieb iMplode nZ ***@***.***>:

**Reply by entropylost** [reply, marked as answer]
Sorry, I feel like I'm misunderstanding something somewhere; you mean that it's the Akinci implementation right? The implementations are mutually exclusive and the other two use implicit fields.

Edit: Never mind, I'll probably just try implementing it myself or something if it isn't the Akinci method.
